src/main/java/dev/slimevr/vr/processor/skeleton/SimpleSkeleton.java

```diff
@@ -704,85 +715,171 @@ protected void updateLocalTransforms() {
+						chestNode.localTransform.setRotation(rotBuf1);
+					} else if (hasKneeTrackers) {
+						// Calculates waist from chest + pelvis
+						leftHipNode.localTransform.getRotation(rotBuf1);
+						rightHipNode.localTransform.getRotation(rotBuf2);
+						chestTracker.getRotation(rotBuf3);
+
+						// Get the rotation relative to where we expect the
+						// upper legs to be
+						rotBuf3.mult(rotBuf4.fromAngles(FastMath.HALF_PI, 0, 0), rotBuf4);
+						if (rotBuf4.dot(rotBuf1) < 0.0f) {
+							rotBuf1.negateLocal();
+						}
+						if (rotBuf4.dot(rotBuf2) < 0.0f) {
+							rotBuf2.negateLocal();
+						}
+
+						rotBuf1.nlerp(rotBuf2, 0.5f);
+						rotBuf3.pureSlerpLocal(rotBuf1, waistChestPelvisFactor);
+						chestNode.localTransform.setRotation(rotBuf3);
+					}
+				}
+				if (hipTracker == null && hasKneeTrackers) {
+					// Calculates hip from (chest or waist) + pelvis
+					leftHipNode.localTransform.getRotation(rotBuf1);
+					rightHipNode.localTransform.getRotation(rotBuf2);
+					getFirstAvailableTracker(waistTracker, chestTracker, null).getRotation(rotBuf3);
+
+					// Get the rotation relative to where we expect the upper
+					// legs to be
+					rotBuf3.mult(rotBuf4.fromAngles(FastMath.HALF_PI, 0, 0), rotBuf4);
+					if (rotBuf4.dot(rotBuf1) < 0.0f) {
+						rotBuf1.negateLocal();
+					}
+					if (rotBuf4.dot(rotBuf2) < 0.0f) {
+						rotBuf2.negateLocal();
+					}
+
+					rotBuf1.nlerp(rotBuf2, 0.5f);
+					rotBuf3.pureSlerpLocal(rotBuf1, hipSpinePelvisFactor);
+					waistNode.localTransform.setRotation(rotBuf3);
+				}
+			}
 		}
 
-		if (extendedPelvisModel && hasKneeTracker) {
+		// Extended pelvis
+		if (extendedPelvisModel && hasKneeTrackers) {
 			// Average pelvis between two legs
 			leftHipNode.localTransform.getRotation(rotBuf1);
 			rightHipNode.localTransform.getRotation(rotBuf2);
 			rotBuf2.nlerp(rotBuf1, 0.5f);
-			chestNode.localTransform.getRotation(rotBuf1);
-			rotBuf2.nlerp(rotBuf1, FastMath.ONE_THIRD);
+			waistNode.localTransform.getRotation(rotBuf1);
+
+			rotBuf2.slerpLocal(rotBuf1, FastMath.ONE_THIRD);
 			hipNode.localTransform.setRotation(rotBuf2);
-			// trackerWaistNode.localTransform.setRotation(rotBuf2); // <==
-			// Provides cursed
-			// results from my test in VRChat when sitting or laying down
-			// -Erimel
-			// TODO : Correct the trackerWaistNode without getting cursed
-			// results (only
-			// correct yaw?)
-			// TODO : Use vectors to add like 50% of waist tracker yaw to waist
-			// node to
-			// reduce drift and let user take weird poses
+
+			// Averages the trackerWaistNode's rotation with the calculated
+			// pelvis'
+			// on local yaw and roll. git blame AxisAngle :p
+			leftHipNode.localTransform.getRotation(rotBuf1);
+			rightHipNode.localTransform.getRotation(rotBuf2);
+			waistNode.localTransform.getRotation(rotBuf3);
+
+			// Get the rotation relative to where we expect the upper legs to be
+			rotBuf3.mult(rotBuf4.fromAngles(FastMath.HALF_PI, 0, 0), rotBuf4);
```

Comment:
Maybe `rotBuf4.fromAngles(FastMath.HALF_PI, 0, 0)` can be a static final Quaternion somewhere to not re-compute every time?